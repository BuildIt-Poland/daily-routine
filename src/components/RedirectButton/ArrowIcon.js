import React from 'react';
import styled from 'styled-components';

import media from '../../styles/media';
import { zIndexDefault } from '../../styles/designTokens';
import { LongArrow } from '../Icons';

const Wrapper = styled.span`
  position: absolute;
  right: 1.4rem;
  bottom: 2.4rem;
  z-index: ${zIndexDefault};

  ${media.phone`
    bottom:  1.4rem;
  `};
`;

function Icon() {
  return (
    <Wrapper>
      <LongArrow />
    </Wrapper>
  );
}

export default Icon;
