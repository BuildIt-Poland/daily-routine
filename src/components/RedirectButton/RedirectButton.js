import React from 'react';
import PropTypes from 'prop-types';

import Button from '../Button';
import ArrowIcon from './ArrowIcon';

function RedirectButton({ children, ...props }) {
  return (
    <Button {...props}>
      {children}
      <ArrowIcon />
    </Button>
  );
}

RedirectButton.propTypes = {
  children: PropTypes.node.isRequired
};

export default RedirectButton;
