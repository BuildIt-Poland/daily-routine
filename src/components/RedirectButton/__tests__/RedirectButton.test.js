import React from 'react';
import { create } from 'react-test-renderer';

import RedirectButton from '../RedirectButton';

jest.mock('react-router-dom', () => ({
  Link: 'Link'
}));
jest.mock('../../Icons', () => ({
  LongArrow: 'LongArrow'
}));

describe('COMPONENT - RedirectButton', () => {
  it('renders correctly', () => {
    const component = create(<RedirectButton to="/taylorswift">Taylor Swift</RedirectButton>);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
