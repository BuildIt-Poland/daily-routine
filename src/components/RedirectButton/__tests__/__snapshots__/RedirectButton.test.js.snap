// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENT - RedirectButton renders correctly 1`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: #21c230;
  color: #ffffff;
  padding: 0 2rem;
  margin: 2rem 0;
  border: 0;
  border-bottom: 0.3rem solid #138a1e;
  border-radius: 0.6rem;
  font-size: 1.8rem;
  font-weight: 700;
  text-align: left;
  width: 100%;
  height: 7rem;
}

.c0:not(:disabled) {
  cursor: pointer;
}

.c1 {
  max-width: 14rem;
}

.c2 {
  position: absolute;
  right: 1.4rem;
  bottom: 2.4rem;
}

@media (max-width:26.25em) {
  .c0 {
    height: 5rem;
  }
}

@media (max-width:26.25em) {
  .c1 {
    max-width: inherit;
  }
}

@media (max-width:26.25em) {
  .c2 {
    bottom: 1.4rem;
  }
}

<Link
  className="c0"
  onClick={[Function]}
  onPointerDown={[Function]}
  onPointerUp={[Function]}
  style={
    Object {
      "borderBottomWidth": 3,
    }
  }
  to="/taylorswift"
>
  <span
    className="c1"
  >
    Taylor Swift
    <span
      className="c2"
    >
      <LongArrow />
    </span>
  </span>
</Link>
`;
